### PR TITLE
fix(view) Ensure current is not null.

### LINKED ISSUE #413: TypeError: current is null.
In aurelia-templating.js, ln 1503, col 63, current is sometimes null. Here is the offending function. 

```
View.prototype.removeNodes = function removeNodes() {
  var fragment = this.fragment;
  var current = this.firstChild;
  var end = this.lastChild;
  var next = void 0;

  while (true) {
    next = current.nextSibling; // bug: current is sometimes null.
    fragment.appendChild(current);
     if (current === end) {
      break;
    }
     current = next;
  }
};
```

A possible fix, of course, is `while(true && current)`.

### PR DESCRIPTION
**Pull request by shaunluttin**
Fixes #413.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
Can you name any scenarios that have triggered this to happen?

**Comment by shaunluttin** [comment]
Yes. It's hard to replicate, though. I could invite you to the private repo to look, if you'd like.

**Comment by EisenbergEffect** [comment]
Is it in conjunction with a 3rd party library, such as a jQuery plugin?

**Comment by shaunluttin** [comment]
It does not appear to be related to jQuery. Our dependencies: 
```

 "jspm": {
    "dependencies": {
      "aurelia-animator-css": "npm:aurelia-animator-css@^1.0.0-rc.1.0.0",
      "aurelia-auth": "npm:aurelia-auth@^3.0.2",
      "aurelia-binding": "npm:aurelia-binding@^1.0.0-rc.1.0.2",
      "aurelia-bootstrapper": "npm:aurelia-bootstrapper@^1.0.0-rc.1.0.1",
      "aurelia-dependency-injection": "npm:aurelia-dependency-injection@^1.0.0-rc.1.0.0",
      "aurelia-event-aggregator": "npm:aurelia-event-aggregator@^1.0.0-rc.1.0.0",
      "aurelia-fetch-client": "npm:aurelia-fetch-client@^1.0.0-rc.1.0.0",
      "aurelia-framework": "npm:aurelia-framework@1.0.0-rc.1.0.1",
      "aurelia-history-browser": "npm:aurelia-history-browser@^1.0.0-rc.1.0.0",
      "aurelia-loader-default": "npm:aurelia-loader-default@^1.0.0-rc.1.0.0",
      "aurelia-logging-console": "npm:aurelia-logging-console@^1.0.0-rc.1.0.0",
      "aurelia-pal-browser": "npm:aurelia-pal-browser@^1.0.0-rc.1.0.0",
      "aurelia-polyfills": "npm:aurelia-polyfills@^1.0.0-rc.1.0.0",
      "aurelia-router": "npm:aurelia-router@^1.0.0-rc.1.0.0",
      "aurelia-templating-binding": "npm:aurelia-templating-binding@^1.0.0-rc.1.0.0",
      "aurelia-templating-resources": "npm:aurelia-templating-resources@^1.0.0-rc.1.0.0",
      "aurelia-templating-router": "npm:aurelia-templating-router@^1.0.0-rc.1.0.0",
      "aurelia-validatejs": "npm:aurelia-validatejs@^0.6.0",
      "aurelia-validation": "npm:aurelia-validation@^0.10.0",
      "bluebird": "npm:bluebird@^3.4.1",
      "fetch": "github:github/fetch@^0.11.0",
      "github/fetch": "github:github/fetch@^1.0.0",
      "jquery": "npm:jquery@^2.2.3",
      "shaunluttin/aurelia-leaflet": "github:shaunluttin/aurelia-leaflet@^0.1.2",
      "text": "github:systemjs/plugin-text@^0.0.3",
      "twbs/bootstrap": "github:twbs/bootstrap@4.0.0-alpha.2"
    },
```

**Comment by EisenbergEffect** [comment]
Ok. I wish I knew what was causing that. It seems like something is removing the "end" node. We can certainly get this in. If you have any more insight as to the root cause, we'd love to hear it though.

**Comment by shaunluttin** [comment]
Sure. I would love to know also. If I find anything, I will update you.

File: src/view.js [hunk 1]
```diff
@@ -216,7 +216,7 @@ export class View {
     let end = this.lastChild;
     let next;
 
-    while (true) {
+    while (true && current) {
```
Comment (EisenbergEffect) [inline src/view.js]: I think on this line we would only need the `while(current)` check. We can remove the `true`. If you can make that change, I'll get this in.

**Comment by shaunluttin** [comment]
BTW... you can watch it happening here: http://sapling-staging.azurewebsites.net

File: src/view.js (hunk 1 above)
Comment (shaunluttin) [inline src/view.js]: Ha. Good point. Silly me.

**Comment by EisenbergEffect** [comment]
@shaunluttin **Thank you for contributing this pull request!** We're truly excited and honored to have you as a member of our community. We look forward to working together with you to further build the Aurelia platform.

We noticed that you haven't yet signed our CLA. We would appreciate it if you would consider doing that for us. Before signing, please take a few minutes to read [our contributing guidelines](https://github.com/DurandalProject/about/blob/master/CONTRIBUTING.md). These guidelines will help get you up to speed on how our community likes to operate and how the pull request process works. Once you have read that, assuming you are in agreement, simply [sign our online CLA](https://docs.google.com/forms/d/1Npd68d32UDUKUYuCDDrW0VN5Lv8p-a6Ziaa2JFKp6Cg/viewform?c=0&w=1).

After you've signed, please ping this issue to let us know and we'll get your improvements merged in.

Thank you again!
The Aurelia Team

**Comment by shaunluttin** [comment]
CLI Signed. :)

**Pull request merged**